src/main/java/fi/aalto/cs/apluscourses/intellij/services/PluginSettings.java

```diff
@@ -15,6 +14,12 @@
 
 public class PluginSettings implements MainViewModelProvider {
 
+  private static PluginSettings instance;
```

Comment:
Possibility of a data race when accessing this?

What about making it non-lazy and constructing the singleton object already when the class is loaded? I think this object is accessed in any case soon after the startup. And constructing it doesn't take much resources.

Reply:
fixed (migrated to eager singleton)

but otherwise, works OK for you:question: @OlliKiljunen

Reply:
Before test was merged to this branch, it worked. Now it gets stuck on "Indexing..." when a new project is created. :(
Hopefully it wasn't my code that broke this.

Reply:
Okay, I can't reproduce that glitch anymore. Perhaps it was some minor bug of IDEA's and as it didn't actually affect anything (it just showed "Indexing..." but menus etc worked and it went away once I imported a module) so sorry for bothering you.

So this works well. 👍 Awesome! 🤩